Source/WebCore/contentextensions/ContentExtensionsBackend.cpp

```diff
@@ -216,8 +219,14 @@ ContentRuleListResults ContentExtensionsBackend::processContentRuleListsForLoad(
     else
         frameURL = url;
 
+    RefPtr<DocumentLoader> mainDocumentLoader;
+    if (RefPtr document = page.mainFrame().document())
+        mainDocumentLoader = document->loader();
+
     ResourceLoadInfo resourceLoadInfo { url, mainDocumentURL, frameURL, resourceType, mainFrameContext };
-    auto actions = actionsForResourceLoad(resourceLoadInfo);
+    auto actions = actionsForResourceLoad(resourceLoadInfo, [loader = mainDocumentLoader](const String& identifier) {
+        return loader && loader->isDisabledContentRuleListIdentifier(identifier) ? ShouldSkipRuleList::Yes : ShouldSkipRuleList::No;
```

Comment:
This is another unnecessary check inside the inner loop.  In fact, this check should be in the same place as what is currently in contentRuleListsEnabled.